tests/storage/linstor/test_linstor_diskless_resources.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+# Requirements:
+# From --hosts parameter:
+# - host(A1): first XCP-ng host >= 8.2 with an additional unused disk for the SR.
+# From --vm parameter
+# - A VM to import to the LINSTOR SR
+# And:
+# - access to XCP-ng RPM repository from hostA1
+
+class TestLinstorDisklessResource:
+    @pytest.mark.small_vm
+    def test_diskless_kept(self, host, linstor_sr, vm_on_linstor_sr):
+        vm = vm_on_linstor_sr
+        vdi_uuids = vm.vdi_uuids()
+        vdi_uuid = None
+        for v in vdi_uuids:
+            if vm.get_vdi_sr_uuid(v) == linstor_sr.uuid:
+                vdi_uuid = v
+                break
+        assert vdi_uuid is not None
+
+        # Get volume name from VDI uuid
+        # "xcp/volume/{vdi_uuid}/volume-name": "{volume_name}"
+        res = host.ssh([
+            "linstor-kv-tool", "--dump-volumes", "-u", "localhost", "-g",
+            "xcp-sr-linstor_group_thin_device", "|", "grep", "volume-name", "|",
+            "grep", f"{vdi_uuid}"
+        ])
+        volume_name = res.split(': ')[1].split('"')[1]
+
+        # Find host where volume is diskless
+        # ┊ {volume_name} ┊ {host} ┊ 7017 ┊ Unused ┊ Ok    ┊   Diskless ┊ 2023-10-24 18:52:05 ┊
+        res = host.ssh([
+            "linstor", "resource", "list", "|", "grep", f"{volume_name}",
+            "|", "grep", "Diskless"
+        ]).splitlines()[0]
+        hostname = res.split('┊')[2].strip()
+
+        # Start VM on host with diskless resource
+        vm.start(on=hostname)
+        vm.wait_for_os_booted()
+
+        # Ensure resource remains diskless
+        res = host.ssh([
+            "linstor", "resource", "list", "|", "grep", f"{volume_name}",
+            "|", "grep", "Diskless", "|", "grep", f"{hostname}"
+        ]).splitlines()
```

Comment:
Why do you split the lines if all you need is testing for the string to be nonempty?